compiler/hash/src/crash_handler.rs

```diff
@@ -0,0 +1,60 @@
+//! Hash compiler crash handler
+//
+// All rights reserved 2021 (c) The Hash Language authors
+
+use backtrace::Backtrace;
+use std::panic::PanicInfo;
+use std::process::exit;
+use std::{io::Write, sync::atomic, thread};
+
+pub(crate) fn panic_handler(info: &PanicInfo) {
+    // keep track to ensure that we only panic once and multiple threads can exit gracefully!
+    static PANIC_ONCE: atomic::AtomicBool = atomic::AtomicBool::new(false);
```

Comment:
So, if two threads panic at the same time, one of them will not be shown. Is that a good idea?

Reply:
I think it is quite unlikely that two threads will panic at the same time, and if they do, what are the chances that they panic on entirely different things.